### PR TITLE
refactor initial queries page

### PR DESCRIPTION
**Pull request by achen2401**
- made changes in the front end logic in initial queries page to accommodate backend logic changes
- refactor initial queries page to retrieve still needed list via API

### CODE REVIEW

File: portal/static/js/main.js [hunk 1]
```diff
@@ -1825,7 +1827,42 @@ OrgTool.prototype.getHereBelowOrgs = function() {
...
+        $.ajax ({
+            type: "GET",
+            url: "/api/coredata/user/" + userId + "/still_needed",
+            cache: false,
+            async: (sync ? false : true)
+        }).done(function(data) {
+            if (data && data.still_needed) {
+                if (callback) callback(data.still_needed);
+            } else {
+                if (callback) callback({"error": "no data returned"});
```
Comment (pbugni) [inline portal/static/js/main.js]: isn't this misleading in the case of a well functioning call, and there's zero items in the still_needed list?

File: portal/templates/initial_queries.html [hunk 2]
```diff
@@ -99,38 +88,76 @@ <h4 class="profile-item-title"> {{ _("Thank you.") }}</h4>
...
-      ma.forEach(function(item) {
-        CONFIG_REQUIRED_CORE_DATA.forEach(function(v) {
-            if (!found && v == item) found = true;
-        });
-      });
-      return found;
-    } else return true;
+/**** object for obtaining required core data ****/
+var configObj = (function() {
+  var CONFIG_DEFAULT_CORE_DATA = {%if config.REQUIRED_CORE_DATA %}{{config.REQUIRED_CORE_DATA|safe}}{% else%}false{%endif%};
```
Comment (pbugni) [inline portal/templates/initial_queries.html]: why is it necessary on the front end to check for this config variable?  this should now always be done via the coredata/api - which considers that, right?

File: portal/templates/initial_queries.html [hunk 3]
```diff
@@ -289,7 +329,7 @@ <h4 class="profile-item-title"> {{ _("Thank you.") }}</h4>
 FieldsChecker.prototype.allFieldsCompleted = function() {
   this.setIncompleteFields();
   var completed = (!hasValue($(".custom-error").text())) && this.incompleteFields.length == 0;
-  if (completed) this.showAll();
+  //if (completed) this.showAll();
```
Comment (pbugni) [inline portal/templates/initial_queries.html]: need to keep this comment?

File: portal/templates/initial_queries.html [hunk 4]
```diff
@@ -525,16 +555,16 @@ <h4 class="profile-item-title"> {{ _("Thank you.") }}</h4>
             $("#month").removeAttr("required").attr("skipped", "true");
             $("#year").removeAttr("required").attr("skipped", "true");
             $(".bd-optional").show();
-           //continueToFinish();
+           //fc.continueToFinish();
```
Comment (pbugni) [inline portal/templates/initial_queries.html]: keep if needed, but looks like dated comments we can remove?

File: portal/static/js/main.js (hunk 1 above)
Comment (achen2401) [inline portal/static/js/main.js]: Paul, I actually need it for the progress bar, which displays whether each core section is completed or not.

File: portal/templates/initial_queries.html (hunk 2 above)
Comment (achen2401) [inline portal/templates/initial_queries.html]: Paul, not sure if my previous comment made it to you.  Typed it quick before the meeting.  I was saying that I actually need it for the progressbar - which shows the state of each section (whether it is completed)

**Comment by achen2401** [comment]
created a new PR, closing this one

**Pull request closed without merging**